library/std/src/collections/hash/set/tests.rs

```diff
@@ -484,3 +484,20 @@ fn test_drain_filter_pred_panic_leak() {
     assert_eq!(DROPS.load(Ordering::SeqCst), 3);
     assert_eq!(set.len(), 0);
 }
+
+#[test]
+fn from_array() {
+    use crate::collections::HashSet;
+
+    let set1 = HashSet::from([1, 2, 3, 4]);
+    let set2: HashSet<_> = [1, 2, 3, 4].into();
+    assert_eq!(set1, set2);
+
+    let unordered_duplicates = HashSet::from([4, 1, 4, 3, 2]);
+    assert_eq!(set1, unordered_duplicates);
```

Comment:
I think you removed too much. I only meant that the last part should be in unit test but this part is still good to show others when they happened to press it in the doc. Having this example is good although it is being shown somewhere else, at least this is more specific.

Reply:
Sure, I've pushed a rebased version of the patch.